ProteoformSuiteInternal/Proteoform.cs

```diff
@@ -185,9 +186,13 @@ public List<ExperimentalProteoform> identify_connected_experimentals(List<PtmSet
                             Lollipop.modification_ranks[m.monoisotopicMass] :
                             Lollipop.modification_ranks[m.monoisotopicMass] + Lollipop.rank_first_quartile / 2; // Penalize modifications that aren't known for this protein and push really rare ones out of the running if they're not in the protein entry
                 }
-                set.ptm_rank_sum = rank_sum + additional_ptm_penalty * (set.ptm_combination.Count - 1); // penalize additional PTMs
+
                 if (rank_sum <= Lollipop.rank_sum_threshold)
```

Comment:
This is the key change. The ptm_rank_sum was changing for the PtmSet objects when constructing them all. Now, it returns a new PtmSet object with the custom rank_sum calculated given the context of the Proteoforms that are connected.